gtsam/hybrid/HybridFactorGraph.cpp

```diff
@@ -58,13 +60,32 @@ const gtsam::DiscreteFactorGraph& HybridFactorGraph::discreteGraph() const {
   return discreteGraph_;
 }
 
+HybridFactorGraph HybridFactorGraph::linearize(
+    const Values& continuousValues) const {
+  // linearize the continuous factors
+  auto gaussian_factor_graph = nonlinearGraph_.linearize(continuousValues);
+
+  // linearize the DCFactors
+  DCFactorGraph linearized_DC_factors;
+  for (DCFactor::shared_ptr factor : dcGraph_) {
+    auto dcgf = factor->linearize(continuousValues);
+    linearized_DC_factors.push_back(dcgf);
+  }
+
+  // Construct new linearized HybridFactorGraph
+  HybridFactorGraph linearizedGraph(this->nonlinearGraph_, this->discreteGraph_,
+                                    linearized_DC_factors,
+                                    *gaussian_factor_graph);
```

Comment:
If the HFG had Gaussian factors, we will lose them here :-) add a check in test that Gaussian factors are left alone.

Reply:
Huh that's a good point. Do we want to append the new gaussian factors to the existing gaussian factor graph? As per this implementation, they will definitely be overwritten,

Reply:
Yes, append.

Reply:
Done